### ISSUE TITLE
`Mv` flag for deletions

### ISSUE THREAD

**Issue by gilhornung** [ChristofferFlensburg/superFreq#30]
Hi Christoffer,

It has been a while... Hope you are doing well.

I am still working on the same project ... reviewing variants.

I came across a group of variants which were flagged with `Mv` (listed below). All of these variants appeared in previous superFreq runs, but failed with this flag in the current run (using superFreq  v0.9.24). The are all deletions (which is suspicious) and also the have `cov` values that are equal to the `var` values, and both values seem wrong. You can see that in the IGV image below. Any idea what could be wrong? What other info do you need?

|Individual |chr |     start|       end|reference                                           |variant |inGene    |Sample | severity|effect  |  f| cov| ref| var|flag | pbq| pmq| psr| somaticScore|germlineLike |dbSNP | dbMAF|dbValidated |ExAC | ExAC_AF|ExAC_Filter                  |varname       |polyPhen |sift |exon |AApos |AAbefore |AAafter |domain |cosmic |isCosmicCensus | cosmicVariantMPM| cosmicGeneMPMPB|
|:----------|:---|---------:|---------:|:---------------------------------------------------|:-------|:---------|:------|--------:|:-------|--:|---:|---:|---:|:----|---:|---:|---:|------------:|:------------|:-----|-----:|:-----------|:----|-------:|:----------------------------|:-------------|:--------|:----|:----|:-----|:--------|:-------|:------|:------|:--------------|----------------:|---------------:|
|WS9        |X   |  17121992|  17122009|NNNNNNNNNNNNNNNNNN                                  |-       |REPS2     |WS9.C  |      100|unknown |  1|   4|   0|   4|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2898155277-18 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |X   |  17121992|  17122009|NNNNNNNNNNNNNNNNNN                                  |-       |REPS2     |WS9.T  |      100|unknown |  1|   3|   0|   3|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2898155277-18 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |X   |  17121992|  17122009|NNNNNNNNNNNNNNNNNN                                  |-       |REPS2     |WS9.R  |      100|unknown |  1|   7|   0|   7|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2898155277-18 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |X   |  30261580|  30261593|NNNNNNNNNNNNNN                                      |-       |MAGEB4    |WS10.R |      100|unknown |  1|  17|   0|  17|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2911294865-14 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |9   |  33045072|  33045072|N                                                   |-       |SMU1      |WS10.R |      100|unknown |  1|  15|   0|  15|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1572204783-1  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |17  |   4758028|   4758030|NNN                                                 |-       |MINK1     |WS10.R |      100|unknown |  1|  14|   0|  14|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2504929891-3  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |1   |  20992614|  20992614|N                                                   |-       |KIF17     |WS10.R |      100|unknown |  1|  12|   0|  12|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |20992613-1    |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |12  | 113729681| 113729699|NNNNNNNNNNNNNNNNNNN                                 |-       |TPCN1     |WS9.C  |      100|unknown |  1|  10|   0|  10|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2064644086-19 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |12  | 113729681| 113729699|NNNNNNNNNNNNNNNNNNN                                 |-       |TPCN1     |WS9.T  |      100|unknown |  1|   6|   0|   6|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2064644086-19 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |12  | 113729681| 113729699|NNNNNNNNNNNNNNNNNNN                                 |-       |TPCN1     |WS9.R  |      100|unknown |  1|  10|   0|  10|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2064644086-19 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |11  |  71279878|  71279907|NNNNNNNNNNNNNNNNNNNNNNNNNNNNNN                      |-       |KRTAP5-10 |WS9.T  |      100|unknown |  1|   8|   0|   8|Mv   |   1|   1|   1|            0|NA           |dbSNP |    NA|TRUE        |ExAC |   0.093|VQSRTrancheINDEL99.50to99.90 |1887187767-30 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |11  |  71279878|  71279907|NNNNNNNNNNNNNNNNNNNNNNNNNNNNNN                      |-       |KRTAP5-10 |WS9.R  |      100|unknown |  1|  16|   0|  16|Mv   |   1|   1|   1|            0|NA           |dbSNP |    NA|TRUE        |ExAC |   0.093|VQSRTrancheINDEL99.50to99.90 |1887187767-30 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |17  |  80050768|  80050768|N                                                   |-       |FASN      |WS9.R  |      100|unknown |  1|   9|   0|   9|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |2580222631-1  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |6   | 127601427| 127601427|N                                                   |-       |RNF146    |WS10.R |      100|unknown |  1|  24|   0|  24|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1190143386-1  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |10  | 105108786| 105108836|NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN |-       |PCGF6     |WS9.C  |      100|unknown |  1|   6|   0|   6|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1785481928-51 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS9        |10  | 105108786| 105108836|NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN |-       |PCGF6     |WS9.R  |      100|unknown |  1|   3|   0|   3|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1785481928-51 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |17  |  33819979|  33819979|N                                                   |-       |SLFN12L   |WS10.C |      100|unknown |  1|   3|   0|   3|Mv   |   1|   1|   1|            0|NA           |dbSNP |    NA|FALSE       |     |      NA|                             |2533991842-1  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |17  |  33819979|  33819979|N                                                   |-       |SLFN12L   |WS10.T |      100|unknown |  1|   9|   0|   9|Mv   |   1|   1|   1|            0|NA           |dbSNP |    NA|FALSE       |     |      NA|                             |2533991842-1  |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |7   | 100065463| 100065472|NNNNNNNNNN                                          |-       |TSC22D4   |WS10.C |      100|unknown |  1|   3|   0|   3|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1333722489-10 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |7   | 100065463| 100065472|NNNNNNNNNN                                          |-       |TSC22D4   |WS10.T |      100|unknown |  1|  14|   0|  14|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |1333722489-10 |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |5   |  33457451|  33457452|NN                                                  |-       |TARS      |WS10.R |      100|unknown |  1|  18|   0|  18|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |915084150-2   |         |     |     |      |         |        |       |       |FALSE          |                0|               0|
|WS10       |1   |    986475|    986481|NNNNNNN                                             |-       |AGRN      |WS10.R |      100|unknown |  1|  16|   0|  16|Mv   |   1|   1|   1|            0|NA           |      |    NA|NA          |     |      NA|                             |986474-7      |         |     |     |      |         |        |       |       |FALSE          |                0|               0|

![image](https://user-images.githubusercontent.com/25452585/36195131-2d5064c6-1175-11e8-88c7-509482347d09.png)

**Comment by gilhornung** [comment]
This part of the code seems to work fine. The total number of reads is not 3:

```
isrv07:.../analysis/curated_pipeline/3-superFreq>samtools mpileup -OsB -f /bio/db/resources/bundle/2.8/b37/human_g1k_v37.fasta -r X:17121991-17121991 ../1-alignment/WS9-T.sorted.markdup.realign.bam
[mpileup] 1 samples in 1 input files
<mpileup> Set max per-file depth to 8000
X       17121991        G       17      .$,.,....-18ATCTGGTCCATGAGATTC.-18ATCTGGTCCATGAGATTC..,-18atctggtccatgagattc,,...       mGmGmmlmmmmGmGmmG       ]]]]]]]]]]]]]]]]] 122,117,110,110,107,104,85,84,83,81,79,75,90,68,66,62,47
isrv07:.../analysis/curated_pipeline/3-superFreq>R-3.2.1

R version 3.2.1 (2015-06-18) -- "World-Famous Astronaut"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-unknown-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> bam = '../1-alignment/WS9-T.sorted.markdup.realign.bam'
> fasta = '/bio/db/resources/bundle/2.8/b37/human_g1k_v37.fasta'
> positions = data.frame(chr='X', start=17121991, reference='G', stringsAsFactors=F)
rary(> library(superFreq)
uperFreq:::bamToPileup(bLoading required package: WriteXLS
am, fasta, positions, index=1)Loading required package: spam
Loading required package: grid
Spam version 1.4-0 (2016-08-29) is loaded.
Type 'help( Spam)' or 'demo( spam)' for a short introduction
and overview of this package.
Help for individual functions is also obtained by adding the
suffix '.spam' to the function name, e.g. 'help( chol.spam)'.

Attaching package: 'spam'

The following objects are masked from 'package:base':

    backsolve, forwardsolve

Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: 'BiocGenerics'

The following objects are masked from 'package:parallel':

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from 'package:spam':

    as.vector, cbind, rbind

The following objects are masked from 'package:stats':

    IQR, mad, xtabs

The following objects are masked from 'package:base':

    Filter, Find, Map, Position, Reduce, anyDuplicated, append,
    as.data.frame, as.vector, cbind, colnames, do.call, duplicated,
    eval, evalq, get, grep, grepl, intersect, is.unsorted, lapply,
    lengths, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unlist, unsplit

Loading required package: GenomicRanges
Loading required package: S4Vectors
Loading required package: stats4

Attaching package: 'stats4'

The following object is masked from 'package:spam':

    mle

Loading required package: IRanges
Loading required package: GenomeInfoDb
Loading required package: biomaRt
Loading required package: Rsamtools
Loading required package: XVector
Loading required package: Biostrings
Loading required package: R.oo
Loading required package: R.methodsS3
R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.
R.oo v1.21.0 (2016-10-30) successfully loaded. See ?R.oo for help.

Attaching package: 'R.oo'

The following object is masked from 'package:GenomicRanges':

    trim

The following object is masked from 'package:IRanges':

    trim

The following objects are masked from 'package:methods':

    getClasses, getMethods

The following objects are masked from 'package:base':

    attach, detach, gc, load, save

Loading required package: Rsubread
Loading required package: limma

Attaching package: 'limma'

The following object is masked from 'package:BiocGenerics':

    plotMA

Loading required package: seqinr

Attaching package: 'seqinr'

The following object is masked from 'package:limma':

    zscore

The following object is masked from 'package:R.oo':

    getName

The following object is masked from 'package:Biostrings':

    translate

The following object is masked from 'package:biomaRt':

    getSequence

The following object is masked from 'package:spam':

    permutation


Attaching package: 'superFreq'

The following object is masked from 'package:limma':

    plotMA

The following object is masked from 'package:BiocGenerics':

    plotMA

> superFreq:::bamToPileup(bam, fasta, positions, index=1)
[mpileup] 1 samples in 1 input files
<mpileup> Set max per-file depth to 8000
$`X:17121991`
   call qual mapq strand note
1     G   76   60      +
2     G   38   60      -
3     G   76   60      +
4     G   38   60      -
5     G   76   60      +
6     G   76   60      +
7     G   75   60      +
8   -18   76   60      +
9   -18   76   60      +
10    G   76   60      +
11    G   76   60      +
12  -18   38   60      +
13    G   76   60      -
14    G   38   60      -
15    G   76   60      +
16    G   76   60      +
17    G   38   60      +

```

**Comment by ChristofferFlensburg** [comment]
Thanks! That all looks fine as you say. So it seems like it's related to the processing of the pileup to read counts that's messed up. Can you quickly show me what you get from (in R)

```
load('myRdirectory/positions.Rdata')
positions['2898155277-18',]
```

(beware of the dash in the variant name that sometimes transforms into other flavours of dashes when copied online)

**Comment by gilhornung** [comment]
The 18bp deletion is not there:
```
> positions['2898155277-18',]
    chr start end  x reference variant inGene
NA <NA>    NA  NA NA      <NA>    <NA>   <NA>
> positions[positions$inGene=="REPS2",]
           chr    start      end          x reference variant inGene
2898076654   X 17043368 17043368 2898076654         C       T  REPS2
2898155277   X 17121991 17121991 2898155277         N      -1  REPS2
> positions[positions$s,]
positions[positions$s
> positions[positions$start==17121992,]
[1] chr       start     end       x         reference variant   inGene
<0 rows> (or 0-length row.names)
```

**Comment by gilhornung** [comment]
I also places the "psotions.Rdata" file and the log in:
https://owncloud.incpm.weizmann.ac.il/index.php/s/tsSZUtR848yWibN
(password as before)

**Comment by ChristofferFlensburg** [comment]
Ok, so the problem is the N in the reference column (the second row is the one we are looking for). SuperFreq now doesn't recognise the Gs as reference reads and interprets them as a different variant on the same position. The deletion then get flagged as the minor variant, as the N->G variant has more support.

I'm pretty sure the variant information is imported from the VCF. Maybe you can double check with a grep on 17121991 (or 17121992) in the input VCF and confirm that there is an N in the reference column. I realise the hypocrisy of superfreq here, struggling with N as reference base for deletions, and then superfreq itself outputs N as reference, but seems like that's how things are at the moment... :P 

I assume you used the same input VCFs in the previous version that worked, so I am not sure why it stopped working. Or rather, I'm not sure why it worked before... But if you can confirm that there is an N in the VCF, and that this worked in a previous version, then I can try to track down what changed.

**Comment by gilhornung** [comment]
OK. Now I understand. I actually do have N in the VCF.

After I experienced a lot of discrepancies between various superFreq runs, I decided I will manually review (using IGV) the somatic mutations that did not repeat between runs. I reviewed 792 mutations (and retained only 24% of them) and then inputed all the curated variants back into superFreq. This explains the Ns...

Thanks a lot for all your help!
